internal/producer/src/accounts/accounts.go

```diff
@@ -277,6 +277,43 @@ func MintAurumUpdateAccountBalanceTable(dbConnection *sql.DB, pkhash []byte, val
 	return errors.New("Failed to find row")
 }
 
+func ValidateContract(c *Contract, table string, authorizedMinters [][]byte) (bool, error) {
+	db, err := sql.Open("sqlite3", table)
+	if err != nil {
+		return false, errors.New("Failed to open table")
+	}
+	defer db.Close()
+
+	// check for zero value transaction
+	if c.Value == 0 {
+		return false, nil
+	}
+
+	// check for unauthorized minting contracts
+	if c.SenderPubKey == nil {
+		validMinting := false
+		for _, mintersPubKHash := range authorizedMinters {
+			if bytes.Equal(c.RecipPubKeyHash, mintersPubKHash) {
+				validMinting = true
```

Comment:
Would it be cleaner to replace the body of this `if` block with lines 307-312, and then put lines 301-304 after the `for` loop?

Reply:
Sure thing